pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -54,99 +27,16 @@ func (e MigrationError) Error() string {
-	logs, err := mg.GetMigrationLog()
-	if err != nil {
-		mg.Logger.Error("Alert migration failure: could not get migration log", "error", err)
-		os.Exit(1)
-	}
-
-	_, migrationRun := logs[migTitle]
-
-	switch {
-	// If unified alerting is enabled and the upgrade migration has not been run
-	case mg.Cfg.UnifiedAlerting.IsEnabled() && !migrationRun:
-		// Remove the migration entry that removes all unified alerting data. This is so when the feature
-		// flag is removed in future the "remove unified alerting data" migration will be run again.
-		mg.AddMigration(fmt.Sprintf(clearMigrationEntryTitle, rmMigTitle), &clearMigrationEntry{
-			migrationID: rmMigTitle,
-		})
-		if err != nil {
-			mg.Logger.Error("Alert migration error: could not clear alert migration for removing data", "error", err)
-		}
-		mg.AddMigration(migTitle, &migration{
-			// We deduplicate for case-insensitive matching in MySQL-compatible backend flavours because they use case-insensitive collation.
-			seenUIDs: uidSet{set: make(map[string]struct{}), caseInsensitive: mg.Dialect.SupportEngine()},
-			silences: make(map[int64][]*pb.MeshSilence),
-		})
-	// If unified alerting is disabled and upgrade migration has been run
-	case !mg.Cfg.UnifiedAlerting.IsEnabled() && migrationRun:
-		// If legacy alerting is also disabled, there is nothing to do
-		if setting.AlertingEnabled != nil && !*setting.AlertingEnabled {
-			return
-		}
-
-		// Safeguard to prevent data loss when migrating from UA to LA
-		if !mg.Cfg.ForceMigration {
-			panic("Grafana has already been migrated to Unified Alerting.\nAny alert rules created while using Unified Alerting will be deleted by rolling back.\n\nSet force_migration=true in your grafana.ini and restart Grafana to roll back and delete Unified Alerting configuration data.")
-		}
-		// Remove the migration entry that creates unified alerting data. This is so when the feature
-		// flag is enabled in the future the migration "move dashboard alerts to unified alerting" will be run again.
-		mg.AddMigration(fmt.Sprintf(clearMigrationEntryTitle, migTitle), &clearMigrationEntry{
-			migrationID: migTitle,
-		})
-		if err != nil {
-			mg.Logger.Error("Alert migration error: could not clear dashboard alert migration", "error", err)
-		}
-		mg.AddMigration(rmMigTitle, &rmMigration{})
-	}
-}
-
-// RerunDashAlertMigration force the dashboard alert migration to run
-// to make sure that the Alertmanager configurations will be created for each organisation
-func RerunDashAlertMigration(mg *migrator.Migrator) {
-	logs, err := mg.GetMigrationLog()
-	if err != nil {
-		mg.Logger.Error("Alert migration failure: could not get migration log", "error", err)
-		os.Exit(1)
-	}
-
+// FixEarlyMigration fixes UA configs created before 8.2 with org_id=0 and moves some files like __default__.tmpl.
+// The only use of this migration is when a user enabled ng-alerting before 8.2.
+func FixEarlyMigration(mg *migrator.Migrator) {
```

Comment:
This used to be called `RerunDashAlertMigration` which was wholly confusing as it didn't actually rerun the migration. Renamed to better represent what it was doing, and removed the condition on migration.

The code is not added to the migration service as it's no longer relevant. See 60d35fbbcf32e0e5d0e73e9da5347ed4d0e6c610